docs/CppInteroperability/ForwardVision.md

```diff
@@ -0,0 +1,290 @@
+
+C++'s types are often either views into managed memory, or managed memory themselves. The line between these two categories is not clear, and neither are the categories themselves. Unlike Objective-C where pointers are almost exclusively managed with ARC, C++ often deals in un-onwed views into managed memory. In the rare case that Objective-C projects it's memory in an un-owned view, the source is to an object with a long and stable lifetime. The fact that C++ leans so heavily on managed pointers and views into these pointers, and that C++'s safety model is so manual, makes C++ interop a much more difficult task that Objective-C interop. 
+
+The detailed approach for importing C++ APIs into Swift follows from the goals above. Much like the example above picked off vairous API patterns from the mess of APIs that `std::vector` provides, C++ interoperability will need to pick off specific API patterns from the mess of all C++ APIs. For example, `std::vector` itself maps well to a Swift value type. The subscript operator can be bridged to a Swift subscript with a getter and a setter. And with some bridging, the `begin` method can be mapped to a safe Swift iterator protocol. The Swift compiler will know about many possible C++ API patterns. If a C++ API has semantic annotations telling Swift that it follows a certain pattern, Swift will try to create a Swift interface for it following the rules of that pattern. In the absence of those annotations, Swift will try to use huristics to recognize an appropriate pattern. If this fails, Swift will make the API unavailable.
+
+One of the most common uses of this "API patterns" concept concerns the import of types. Swift types fall into two categories: value types and reference types. Copying a value of a reference type produces a new reference to the same underlying object, similar to an intrusive `std::shared_ptr` in C++ or a class type in Java. Copying a value of a value type recursively copies the components of the type, similar to the behavior of a struct in C or the default behavior of a class type in C++. Furthermore, both kinds of types must always be copyable, although there are plans in the works to allow types to restrict this.
```

Comment:
Okay, so we pick up here with this paragraph after the suggestion above.  I'll provide more feedback starting from here tomorrow.